@navikt/aksel-icons/config/figma/index.ts

```diff
@@ -28,57 +28,58 @@ async function main() {
 
   mkdirSync(iconFolder);
 
-  console.info(
-    `Downloading ${Object.keys(imagesUrls).length} icons from Figma...`,
-  );
+  console.group(`Processing ${Object.keys(imagesUrls).length} icons...`);
 
-  let counter = 0;
-  for (const [nodeId, iconUrl] of Object.entries(imagesUrls)) {
-    const iconSvg = await fetch(iconUrl)
-      .then((x) => x.text())
-      .catch((e) => {
-        throw e.message;
-      });
+  const fileNames = new Set<string>();
 
-    if (!iconSvg) {
-      continue;
-    }
+  await Promise.all(
+    Object.entries(imagesUrls).map(async ([nodeId, iconUrl]) => {
+      const iconSvg = await fetch(iconUrl)
+        .then((x) => x.text())
+        .catch((e) => {
+          throw e.message;
+        });
 
-    /*
-     * Arbitrary delay to not get rate-limited by image hosting
-     * Currently accounts for ~18 seconds in theory, but in practice the bottlenech is fetching each icon 1 at a time
-     */
-    await new Promise((r) => setTimeout(r, 20));
+      const matchingIcon = publishedIconComponents.find(
+        (x) => x.node_id === nodeId,
+      );
 
-    counter++;
+      if (!matchingIcon) {
+        throw new Error(
+          `No matching icon found for ${nodeId}. It should not be possible to dowload icon without a matching icon in the list of icons fetched from Figma.`,
+        );
+      }
 
-    if (counter % 20 === 0) {
-      process.stdout.write(`Processed ${counter} icons\r`);
-    }
+      const fileName = resolveName(matchingIcon);
 
-    const matchingIcon = publishedIconComponents.find(
-      (x) => x.node_id === nodeId,
-    );
+      if (fileNames.has(fileName)) {
+        console.warn(`Duplicate name detected: ${fileName}.`);
+      }
 
-    if (!matchingIcon) {
-      throw new Error(
-        `No matching icon found for ${nodeId}. It should not be possible to dowload icon without a matching icon in the list of icons fetched from Figma.`,
-      );
-    }
+      fileNames.add(fileName);
 
-    writeFileSync(resolve(iconFolder, resolveName(matchingIcon)), iconSvg, {
-      encoding: "utf8",
-    });
+      writeFileSync(resolve(iconFolder, resolveName(matchingIcon)), iconSvg, {
+        encoding: "utf8",
+      });
+    }),
+  );
+
+  if (fileNames.size !== publishedIconComponents.length) {
+    console.warn(
+      `Duplicate icon names from Figma leads to them being overwritten. This will cause the icon-library to be out of sync with Figma.`,
+    );
   }
-  console.info(`Completed processing of ${counter} icons`);
+
+  console.info(`Completed processing 🎉`);
+  console.groupEnd();
 
   makeConfig(publishedIconComponents, iconFolder);
 
   const filesInDir = readdirSync(iconFolder);
 
   if (filesInDir.length * 2 !== publishedIconComponents.length) {
```

Comment:
Is this math correct? Shouldn't you divide by two to get the number of icons? You could also just filter by svg first, maybe that's clearer.

Reply:
Fixed